### PR TITLE
Fix/no floating promises

### LINKED ISSUE #3515: Reuires Internet?
Using the vscode extension (installed from vscode) rust analyzer fails to load. If I connect to wifi I have no problems. However if I connect to wifi open vscode (rust analyzer works correctly at this point) and then disconnect from wifi, close and reopen vscode then I get the same error. 

Without looking through the code base I would _guess_ that the extension is looking for a newer version of rust analyzer and when it fails to connect to the internet the entire extension fails.

### PR DESCRIPTION
**Pull request by sahandevs**
closes #3515

- added `@typescript-eslint/no-floating-promises: error` rule
- changed `"no-console": ["error"]` to `"no-console": ["error", { allow: ["warn", "error"] }]` (we at least log the error messages of the floating promises)
- fixed lint/compile errors

### CODE REVIEW

File: editors/code/src/commands.ts [hunk 1]
```diff
@@ -129,7 +129,8 @@ export function joinLines(ctx: Ctx): Cmd {
             client.protocol2CodeConverter.asTextEdits(items).forEach((edit: any) => {
                 builder.replace(edit.range, edit.newText);
             });
-        });
+        })
+            .then(() => { }, console.error);
```
Comment (bjorn3) [inline editors/code/src/commands.ts:133]: Doesn't this prevent vscode from reporting the error to the user?

File: editors/code/src/commands.ts [hunk 2]
```diff
@@ -477,7 +480,8 @@ export function applyActionGroup(_ctx: Ctx): Cmd {
         vscode.commands.executeCommand(
             'rust-analyzer.resolveCodeAction',
             selectedAction.arguments,
-        );
+        )
+            .then(() => { }, console.error);
```
Comment (matklad) [inline editors/code/src/commands.ts:484]: Here and elsewhere, seems like we can `await`? Actually fixing the problem is preferable to sweeping it under the rug ;)

Comment (sahandevs) [reply editors/code/src/commands.ts:484]: @matklad doesn't this require us to change all of the parent function signatures to async?

Comment (matklad) [reply editors/code/src/commands.ts:484]: parent function here is already async, no?

Comment (sahandevs) [reply editors/code/src/commands.ts]: @matklad I made some few changes, can u check again? also I noticed there is a type called `Cmd` which is a normal function but we are using it as if we are returning an async function (example: `commands.ts/analyzerStatus()`) and at the end we are calling them like normal functions so we don't `await` or `catch() `them. doesn't this cause the same trouble?

**Pull request merged**